moshi/src/test/java/com/squareup/moshi/AdapterMethodsTest.java

```diff
@@ -0,0 +1,131 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.moshi;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public final class AdapterMethodsTest {
+  @Test public void toAndFromJsonViaListOfIntegers() throws Exception {
+    Moshi moshi = new Moshi.Builder()
+        .add(new PointAsListOfIntegersJsonAdapter())
+        .build();
+    JsonAdapter<Point> pointAdapter = moshi.adapter(Point.class);
+    assertThat(pointAdapter.toJson(new Point(5, 8))).isEqualTo("[5,8]");
+    assertThat(pointAdapter.fromJson("[5,8]")).isEqualTo(new Point(5, 8));
+  }
+
+  static class PointAsListOfIntegersJsonAdapter {
+    @ToJson List<Integer> pointToJson(Point point) {
+      return Arrays.asList(point.x, point.y);
+    }
+
+    @FromJson Point pointFromJson(List<Integer> o) throws Exception {
+      if (o.size() != 2) throw new Exception("Expected 2 elements but was " + o);
+      return new Point(o.get(0), o.get(1));
+    }
+  }
+
+  @Test public void toJsonOnly() throws Exception {
+    Moshi moshi = new Moshi.Builder()
+        .add(new PointAsListOfIntegersToAdapter())
+        .build();
+    JsonAdapter<Point> pointAdapter = moshi.adapter(Point.class);
+    assertThat(pointAdapter.toJson(new Point(5, 8))).isEqualTo("[5,8]");
+    assertThat(pointAdapter.fromJson("{\"x\":5,\"y\":8}")).isEqualTo(new Point(5, 8));
+  }
+
+  static class PointAsListOfIntegersToAdapter {
+    @ToJson List<Integer> pointToJson(Point point) {
+      return Arrays.asList(point.x, point.y);
+    }
+  }
+
+  @Test public void fromJsonOnly() throws Exception {
+    Moshi moshi = new Moshi.Builder()
+        .add(new PointAsListOfIntegersFromAdapter())
+        .build();
+    JsonAdapter<Point> pointAdapter = moshi.adapter(Point.class);
+    assertThat(pointAdapter.toJson(new Point(5, 8))).isEqualTo("{\"x\":5,\"y\":8}");
+    assertThat(pointAdapter.fromJson("[5,8]")).isEqualTo(new Point(5, 8));
+  }
+
+  static class PointAsListOfIntegersFromAdapter {
+    @FromJson Point pointFromJson(List<Integer> o) throws Exception {
+      if (o.size() != 2) throw new Exception("Expected 2 elements but was " + o);
+      return new Point(o.get(0), o.get(1));
+    }
+  }
+
+  @Test public void multipleLayersOfAdapters() throws Exception {
+    Moshi moshi = new Moshi.Builder()
+        .add(new MultipleLayersJsonAdapter())
+        .build();
+    JsonAdapter<Point> pointAdapter = moshi.adapter(Point.class).lenient();
+    assertThat(pointAdapter.toJson(new Point(5, 8))).isEqualTo("\"5 8\"");
+    assertThat(pointAdapter.fromJson("\"5 8\"")).isEqualTo(new Point(5, 8));
+  }
+
+  static class MultipleLayersJsonAdapter {
```

Comment:
I'm struggling to figure out when this would be useful. Why wouldn't you just go directly to the desired JSON format?